libp2p/muxers/mplex/mplex.nim

```diff
@@ -190,17 +197,30 @@ method close*(m: Mplex) {.async, gcsafe.} =
   if m.isClosed:
     return
 
-  trace "closing mplex muxer", oid = m.oid
-
-  checkFutures(
-    await allFinished(
-      toSeq(m.remote.values).mapIt(it.reset()) &
-        toSeq(m.local.values).mapIt(it.reset())))
-
-  checkFutures(await allFinished(m.handlerFuts))
-
-  await m.connection.close()
-  m.remote.clear()
-  m.local.clear()
-  m.handlerFuts = @[]
-  m.isClosed = true
+  try:
+    trace "closing mplex muxer", oid = m.oid
+    let channs = toSeq(m.remote.values) &
+      toSeq(m.local.values)
+
+    for chann in channs:
```

Comment:
well looks like `checkFutures` / `allFinished` with a added `warn` mode would be fine here

Reply:
I don't want to stay away from them in this parts, because they've shown to be the most problematic, eventually we'll read them, but for now I want to keep them as is.

Reply:
this would be a good time to put a comment on every place where the error is ignored, why it makes sense to do so - ie document the rationale - this would need to be done every place it happens so as to help future refactoring efforts and new code to follow the same "structure" of error handling

Reply:
basically, if the strategy is "ignore write errors", there should be a comment pointing out where more information about the error and ownership flow of errrors-during-write happens